app/models/service.rb

```diff
@@ -211,6 +209,14 @@ def power_states_match?(action)
     false
   end
 
+  def composite?
+    service_template ? service_template.service_type.to_s.include?('composite') : children.present?
```

Comment:
If you have a service_template all you have to do is call the existing method on that class.  
`service_template ? service_template.composite? : children.present?`  (Same for `atomic?` below)

@kbrock Is `children.present?` going to load all the child objects?  Is there a way to check for children in ancestry without the overhead of loading them?